fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/FabricStructurePool.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.structure.v1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.mojang.datafixers.util.Pair;
+
+import net.minecraft.structure.pool.StructurePool;
+import net.minecraft.structure.pool.StructurePoolElement;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.mixin.structure.StructurePoolAccessor;
+
+/**
+ * Represents a modifiable structure pool that has several helper methods for modders.
+ */
+public record FabricStructurePool(StructurePool underlying, Identifier id) {
```

Comment:
Is a record really the right choice here? An interface would be better imo:

- A record (and interface) shouldnt really contain logic
- It makes it easier for someone else (or another module) to re-impliment the API.
- It makes the API surface clear, I dont think its too bad here but would be better as a simple interface.

The impl could still be a record (if it makes sense) but the exposed API should really be an interface imo.